universe/Ship.cpp

```diff
@@ -590,6 +589,7 @@ void Ship::SetShipMetersToMax() {
         if (max_it == m_part_meters.end())
             continue;
 
+        max_it->second.SetCurrent(Meter::LARGE_VALUE);
```

Comment:
After more mulling over why things don't work without the above code line I had some idea: Could it be that, when only the current value of the capacity meter is set, a clamping occurs *before* any effects are applied to the max capacity meter? In that case the capacity meter would be reset to the default initial value of the max meter (which is 0 AFAIK). Then the effects which set the max meter are applied, followed by another clamping(s), but that wouldn't do any good, as the capacity meter has already been reduced to 0.

And from what I've seen, there is no place where the weapon part capacity meters ever get adjusted to the max meters (with exactly one exception, the resupply code, which is why empire owned ships get their weapon strength corrected), so they stay at 0.

With the above code line in place however, a clamping occurring before the max meter is correctly set by effects will not reduce the capacity meter to 0, but stay at `Meter:LARGE_VALUE` , and further clampings subsequent to effects execution will set the capacity meter to the correct max.

That's my guess...